classes/local/autocomplete_suggestion_helper.php

```diff
@@ -0,0 +1,211 @@
+        $suggestionlist = [];
+
+        // Prepare the select sql with join.
+        $sql = "SELECT uj.id, uj.userid, md.metadata FROM {block_opencast_uploadjob} uj "
+            ."JOIN {block_opencast_metadata} md ON uj.id = md.uploadjobid "
+            ."WHERE uj.courseid = :courseid";
+        
+        $params = [];
+        $params['courseid'] = $COURSE->id;
+
+        // Get records from db.
+        $records = $DB->get_records_sql($sql, $params);
+
+        // If no records, we return empty array.
+        if (!$records) {
+            return $suggestionlist;
+        }
+
+        foreach ($records as $record) {
+            // At first, we consider the user who upload the video as an suggestion.
+
+            // Get the user who performed the upload job.
+            $uploadjobuser = \core_user::get_user($record->userid);
+
+            // Get the fullname of the user.
+            $uploadjobuserfullname = fullname($uploadjobuser);
+
+            // Check if the fullname is valid or if it is not already existed.
+            if (!empty($uploadjobuserfullname) && !in_array($uploadjobuserfullname, $suggestionlist)) {
+                $suggestionlist[] = $uploadjobuserfullname;
+            }
+
+            // Next, we check the stored metadada to see if there is any creator or contributor fields.
+
+            // Get metadata.
+            $metadata = json_decode($record->metadata);
+
+            // Check and return the creator field from stored metadada catalog.
+            $creatorarray = array_filter($metadata, function($v, $k) {
+                return $v->id == 'creator';
+            }, ARRAY_FILTER_USE_BOTH);
+
+            // If there is creator field in stored metadada catalog.
+            if (!empty($creatorarray)) {
+                // Extract the creator object from filtered array.
+                $creatormetadataobject = array_values($creatorarray)[0];
+
+                // Merge the values of the object with the returned list.
+                $suggestionlist = array_merge($suggestionlist, $creatormetadataobject->value);
+            }
+
+            // Check and return the contributor field from stored metadada catalog.
+            $contributorarray = array_filter($metadata, function($v, $k) {
+                return $v->id == 'contributor';
+            }, ARRAY_FILTER_USE_BOTH);
+
+            // If there is contributor field in stored metadada catalog.
+            if (!empty($contributorarray)) {
+                // Extract the contributor object from filtered array.
+                $contributormetadataobject = array_values($contributorarray)[0];
+
+                // Merge the values of the object with the returned list.
+                $suggestionlist = array_merge($suggestionlist, $contributormetadataobject->value);
+            }
+        }
+
+        // Finally, we remove the duplicated values from the return list and return it.
+        return array_unique($suggestionlist);
```

Comment:
You don't need `array_unique` here because you are calling it in `get_suggestions_for_creator_and_contributer` anyway. It's not a big deal but if we can optimize the code and make it faster, it's always a good idea.